src/test/java/zmq/HeartbeatsTest.java

```diff
@@ -16,6 +16,50 @@
 
 public class HeartbeatsTest
 {
+    // this checks for heartbeat in REQ socket.
+    @Test
+    public void testHeartbeatReq() throws IOException
+    {
+        final int heartbeatInterval = 100;
+        final int port = Utils.findOpenPort();
+
+        final String addr = String.format("tcp://localhost:%s", port);
+
+        Ctx ctx = ZMQ.init(1);
+        assertThat(ctx, notNullValue());
+
+        SocketBase req = ctx.createSocket(ZMQ.ZMQ_REQ);
+        assertThat(req, notNullValue());
+        boolean rc = req.setSocketOpt(ZMQ.ZMQ_HEARTBEAT_IVL, heartbeatInterval);
+        assertThat(rc, is(true));
+        rc = req.connect(addr);
+        assertThat(rc, is(true));
+
+        SocketBase rep = ctx.createSocket(ZMQ.ZMQ_REP);
+        assertThat(rep, notNullValue());
+        rc = rep.bind(addr);
+        assertThat(rc, is(true));
+
+        final long start = System.currentTimeMillis();
+        do {
+            // request
+            int ret = ZMQ.send(req, "hello", 0);
+            assertThat(ret, is(5));
```

Comment:
It isn't clear what this value 5 means. It would be good to see a named constant, so that it is self-documenting.